main.md

```diff
@@ -364,177 +390,47 @@ The client has sent an authorization request using the following example authori
-         ],
-         "locations": [
-            "https://example.com/payments"
-         ],
-         "instructedAmount": {
-            "currency": "EUR",
-            "amount": "123.50"
-         },
-         "creditorName": "Merchant123",
-         "creditorAccount": {
-            "iban": "DE02100100109307118603"
-         },
-         "remittanceInformationUnstructured": "Ref Number Merchant"
-      }
-   ],
-   ...
-}
-```
+# Authorization Response
 
-# Using "authorization_details"
+This specification does not define extensions to the authorization response. 
 
-## Authorization Request
+# Authorization Error Response
 
-The request parameter can be used to specify authorization requirements in all places where the `scope` parameter is used for the same purpose, examples include:  
-
-* Authorization requests as specified in [@!RFC6749], 
-* Access token requests as specified in [@!RFC6749], if also used as authorization requests, e.g. in the case of assertion grant types [@!RFC7521],
-* Request objects as specified in [@I-D.ietf-oauth-jwsreq], 
-* Device Authorization Request as specified in [@!RFC8628],
-* Backchannel Authentication Requests as defined in [@OpenID.CIBA].
+The AS MUST refuse to process any unknown authorization data type or authorization details not conforming to the respective type definition. If any of the objects in `authorization_details` contains an unknown authorization data type or an object of 
+known type but containing unknown elements or elements of the wrong type, the AS MUST abort processing and respond with an error `invalid_authorization_details` to the client. 
 
-Parameter encoding is determined by the respective context. 
+# Token Request
 
-In the context of an authorization request according to [@!RFC6749], the parameter is encoded using the `application/x-www-form-urlencoded` format of the serialized JSON as shown in the following example:
-
-```
-GET /authorize?response_type=code
-   &client_id=s6BhdRkqt3
-   &state=af0ifjsldkj
-   &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb
-   &code_challenge_method=S256
-   &code_challenge=K2-ltc83acc4h0c9w6ESC_rEMTJ3bwc-uCHaoeK1t8U
-   &authorization_details=%5B%7B%22type%22%3A%22account%5Finformati
-   on%22%2C%22actions%22%3A%5B%22list%5Faccounts%22%2C%22read%5Fbal
-   ances%22%2C%22read%5Ftransactions%22%5D%2C%22locations%22%3A%5B%
-   22https%3A%2F%2Fexample%2Ecom%2Faccounts%22%5D%7D%5D HTTP/1.1
-Host: server.example.com
-``` 
-
-Implementors MUST ensure to protect personal identifiable information
-in transit. One way is to utilize encrypted request objects as defined
-in [@I-D.ietf-oauth-jwsreq]. In the context of a request object, 
-`authorization_details` is added as another top level JSON element.
+The `authorization_details` request parameter MAY be used in a token request to further restrict the permissions assigned to an access token. As an example, the grant approved in the running example above could be restricted as follows
```

Comment:
I really dislike this and don't think the value is worth the complexity.

Reply:
While restructuring it become obvious to me that we did not specify so far whether and how the authorization_details parameter could be used at the token request. We had comments on the list asking for it to downgrade the permissions associated to an access token. I don't have a use case for it and could also live with removing it again. 
@jricher what do you think?

Reply:
I'm somewhat skeptical of the actual need for downgrade/scope. But if there is a need, I'd like to consider other ways of addressing it that might be more straightforward than application specific constructs that have to determine if one chunk of JSON is a proper down-scoped subset of another. That sounds very hard to implement and a likely vector for accidentally allowing for up-scoping beyond the actual granted authz.

Reply:
I get why it's there, it's in parallel to how `scope` can be used and so that much makes sense. However, `scope` is much, much simpler to describe what "downgrade" means: you remove one of the scope values from the list, probably. With RAR it's a much more complicated calculation for the AS to determine if it's "less" or not. But even then, as long as the AS fails "closed" in cases where it can't tell, and it denies the request, then it should be OK in practice. 

What if the rubric was instead "would not require additional consent from the RO"? Or is that getting to deep into UX territory?

Reply:
"What if the rubric was instead "would not require additional consent from the RO"? "

What would change then? If I understand it right, the AS would still need to determine automatically whether the authz details match already consented authz details.

Reply:
I thought that no additional consent was kinda implied by the "to further restrict the permissions assigned" already. 
My push-back on the feature is about the added implementation complexity (with potential for introducing security issues) vs. actual utility of it.

Reply:
I removed this paragraph. So authorization_details is authz request only now and resource is token request only. Are we good to merge now?